test_multiprocessing broken with default multiprocessing set to "spawn"

When the default multiprocessing context is `'spawn'` rather than `'fork'`, `tests/test_sync.py::test_multiprocessing`. This is e.g. the case on Darwin and it might be the case in final PyPy3.9 release because forking multithreaded processes causes frequent deadlocks.

To easily reproduce this problem using regular CPython, you can modify `/usr/lib/python3.*/multiprocessing/context.py`, changing:

```python
_default_context = DefaultContext(_concrete_contexts['fork'])
```
to:

```python
_default_context = DefaultContext(_concrete_contexts['spawn'])
```

Traceback (from CPython 3.10 modified as above):

```python
________________________________________________________ test_multiprocessing _________________________________________________________

    @pytest.mark.asyncio
    async def test_multiprocessing():
        """
        Tests that a forked process can use async_to_sync without it looking for
        the event loop from the parent process.
        """
    
        test_queue = multiprocessing.Queue()
    
        async def async_process():
            test_queue.put(42)
    
        def sync_process():
            """Runs async_process synchronously"""
            async_to_sync(async_process)()
    
        def fork_first():
            """Forks process before running sync_process"""
            fork = multiprocessing.Process(target=sync_process)
            fork.start()
            fork.join(3)
            # Force cleanup in failed test case
            if fork.is_alive():
                fork.terminate()
            return test_queue.get(True, 1)
    
>       assert await sync_to_async(fork_first)() == 42

tests/test_sync.py:636: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
asgiref/sync.py:414: in __call__
    ret = await asyncio.wait_for(future, timeout=None)
/usr/lib/python3.10/asyncio/tasks.py:408: in wait_for
    return await fut
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
asgiref/sync.py:455: in thread_handler
    return func(*args, **kwargs)
tests/test_sync.py:629: in fork_first
    fork.start()
/usr/lib/python3.10/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
/usr/lib/python3.10/multiprocessing/context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
/usr/lib/python3.10/multiprocessing/context.py:284: in _Popen
    return Popen(process_obj)
/usr/lib/python3.10/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
/usr/lib/python3.10/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
/usr/lib/python3.10/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <Process name='Process-1' parent=1279098 initial>, file = <_io.BytesIO object at 0x7f5e8d433ce0>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       AttributeError: Can't pickle local object 'test_multiprocessing.<locals>.sync_process'

/usr/lib/python3.10/multiprocessing/reduction.py:60: AttributeError
```

--- a/tests/test_sync.py
+++ b/tests/test_sync.py
@@ -607,32 +607,33 @@ def test_sync_to_async_detected_as_coroutinefunction():
     assert asyncio.iscoroutinefunction(sync_to_async(sync_func))
 
 
+async def async_process(queue):
+    queue.put(42)
+
+
+def sync_process(queue):
+    """Runs async_process synchronously"""
+    async_to_sync(async_process)(queue)
+
+
+def fork_first():
+    """Forks process before running sync_process"""
+    queue = multiprocessing.Queue()
+    fork = multiprocessing.Process(target=sync_process, args=[queue])
+    fork.start()
+    fork.join(3)
+    # Force cleanup in failed test case
+    if fork.is_alive():
+        fork.terminate()
+    return queue.get(True, 1)
+
+
 @pytest.mark.asyncio
 async def test_multiprocessing():
     """
     Tests that a forked process can use async_to_sync without it looking for
     the event loop from the parent process.
     """
-
-    test_queue = multiprocessing.Queue()
-
-    async def async_process():
-        test_queue.put(42)
-
-    def sync_process():
-        """Runs async_process synchronously"""
-        async_to_sync(async_process)()
-
-    def fork_first():
-        """Forks process before running sync_process"""
-        fork = multiprocessing.Process(target=sync_process)
-        fork.start()
-        fork.join(3)
-        # Force cleanup in failed test case
-        if fork.is_alive():
-            fork.terminate()
-        return test_queue.get(True, 1)
-
     assert await sync_to_async(fork_first)() == 42
 
 

